pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UseCollectionIsEmptyRule.java

```diff
@@ -112,11 +113,14 @@ private String getVariableNameBySuffix(ASTPrimaryExpression primaryExpression) {
     }
 
     private ASTClassOrInterfaceType getTypeOfVariableByName(String varName, ASTPrimaryExpression expr) {
-        ASTClassOrInterfaceBody classBody = expr.getFirstParentOfType(ASTClassOrInterfaceBody.class);
-        List<ASTVariableDeclarator> varDeclarators = classBody.findDescendantsOfType(ASTVariableDeclarator.class);
+        Node classOrEnumBody = expr.getFirstParentOfType(ASTClassOrInterfaceBody.class);
+        if (classOrEnumBody == null) {
+            classOrEnumBody = expr.getFirstParentOfType(ASTEnumBody.class);
+        }                 
+        List<ASTVariableDeclarator> varDeclarators = classOrEnumBody.findDescendantsOfType(ASTVariableDeclarator.class);
         for (ASTVariableDeclarator varDeclarator : varDeclarators) {
             if (varDeclarator.getName().equals(varName)) {
-                return varDeclarator.getFirstDescendantOfType(ASTClassOrInterfaceType.class);
+                return varDeclarator.getParent().getFirstDescendantOfType(ASTClassOrInterfaceType.class);
```

Comment:
What's the reason to go one level up in the hierarchy?

Reply:
This fetches a LocalVariableDeclaration or FieldDeclaration node. VariableDeclarator nodes never have themselves a type node child.

But the actual right way to do that is `varDeclarator.getVariableId().getTypeNode()`.

Reply:
This fetches a LocalVariableDeclaration or FieldDeclaration node. VariableDeclarator nodes never have themselves a type node child.

But the actual right way to do that is `varDeclarator.getVariableId().getTypeNode()`.

Reply:
I'll fix that when merging